app/src/main/java/com/arctouch/codechallenge/repository/implementation/SearchMoviesRepositoryImplementation.java

```diff
@@ -0,0 +1,58 @@
+package com.arctouch.codechallenge.repository.implementation;
+
+import com.arctouch.codechallenge.api.TmdbApi;
+import com.arctouch.codechallenge.api.TmdbApiClient;
+import com.arctouch.codechallenge.callback.OnGetSearchMovies;
+import com.arctouch.codechallenge.entity.UpcomingMoviesResponse;
+import com.arctouch.codechallenge.repository.SearchMoviesRepository;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class SearchMoviesRepositoryImplementation implements SearchMoviesRepository{
+
+    private TmdbApi service;
+
+    public SearchMoviesRepositoryImplementation() {
```

Comment:
When you return to `HomeActivity` from search, the results are not reseting.

Reply:
Yes, and this is not the only thing. I tried to implement and deliver everything that was requested with the time I had, but like I mentioned, I would have gone with a better implementation of the search functionality. 
A few known issues with it so far:
- results are not reseting, like you pointed out ,
- if you erase search, infinite scrolling will start adding common results to the searched ones,
- keyboard needs a better treatment (when scrolling, dismissing properly, etc),
- you can swipe to refresh on search results - overall results will overwrite them

But still, tried to deliver some search capabilities. Hope that counts!